src/lib.rs

```diff
@@ -973,3 +976,16 @@ fn getenv_unwrap(v: &str) -> String {
 fn fail(s: &str) -> ! {
     panic!("\n{}\n\nbuild script failed, must exit now", s)
 }
+
+// Returns the major, minor and patch versions of cmake
+fn cmake_version(executable: &OsStr) -> io::Result<(u8, u8, u8)> {
+    let cmd = Command::new(executable).arg("--version").output()?;
+    let version = String::from_utf8_lossy(&cmd.stdout);
+    let version: Vec<&str> = version.split_whitespace().collect();
+    let version: Vec<&str> = version[2].split('.').collect();
```

Comment:
I think the indexing here should be fallible and I don't think that parse errors below should be papered over with 0.